src/luarocks/fs/win32/tools.lua

```diff
@@ -238,16 +244,19 @@ function tools.set_permissions(filename, mode, scope)
       if not ok then
          return false, "Could not take ownership of the given file"
       end
+      local username = os.getenv('USERNAME')
       -- Grant the current user the proper rights
-      ok = fs.execute_quiet(vars.ICACLS .. " " .. fs.Q(filename) .. " /inheritance:d /grant:r \"%USERNAME%\":" .. perms)
+      ok = fs.execute_quiet(vars.ICACLS .. " " .. fs.Q(filename) .. " /inheritance:d /grant:r " .. fs.Q(username) .. ":" .. perms)
```

Comment:
do you need `fs.Q(vars.ICACLS)`?

Reply:
This is not my change. This was remove somwhere between 3.0.4 and 3.5.0

Reply:
We keep the command name variables unquoted so that a user can override them adding custom flags, e.g. `ICACLS = "icacls /mycustomflag"`